### ISSUE TITLE
Breast Cancer_Sample Example

### ISSUE THREAD

**Issue by zhangling297** [zhangling297/Substance-Use#3]
Dua, D. and Graff, C. (2019). UCI Machine Learning Repository [[http://archive.ics.uci.edu/ml]](http://archive.ics.uci.edu/ml%5D). Irvine, CA: University of California, School of Information and Computer Science.

Breast Cancer Wisconsin (Diagnostic) Data Set [[https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Diagnostic)]](https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Diagnostic)%5D).

Also see: Breast Cancer Wisconsin (Diagnostic) Data Set [[https://www.kaggle.com/uciml/breast-cancer-wisconsin-data]](https://www.kaggle.com/uciml/breast-cancer-wisconsin-data%5D).

Let's download the data and save it in the local folder with the name data.csv and take a look at it.

Breast Cancer Prediction
Predict Breast Cancer with features derived from images of Breast Mass

Contents
[Background](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Background)
[Setup](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Setup)
[Data](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Data)
[Train](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Train)
[Host](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Host)
[Predict](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Predict)
[Extensions](https://basicnotebookinstance-ki5dztuymuwg.notebook.us-east-1.sagemaker.aws/notebooks/Access-to-Data-and-Compute-using-Service-Workbench_2025-10-29/Breast%20Cancer%20Prediction%20on%20SWB%20-%20python.ipynb#Extensions)

Background
This notebook illustrates how one can use random forest models for prediction. For this illustration, we have taken an example for breast cancer prediction using UCI'S breast cancer diagnostic data set. The purpose here is to use this data set to build a predictve model of whether a breast mass image indicates benign or malignant tumor.

import os
import boto3
import re
import sagemaker
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import io
import time
import json
import sagemaker.amazon.common as smac
import pathlib
import os
print(os.path.abspath(os.getcwd()))

Data sources
Dua, D. and Graff, C. (2019). UCI Machine Learning Repository [[http://archive.ics.uci.edu/ml]](http://archive.ics.uci.edu/ml%5D). Irvine, CA: University of California, School of Information and Computer Science.

Breast Cancer Wisconsin (Diagnostic) Data Set [[https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Diagnostic)]](https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Diagnostic)%5D).

Also see: Breast Cancer Wisconsin (Diagnostic) Data Set [[https://www.kaggle.com/uciml/breast-cancer-wisconsin-data]](https://www.kaggle.com/uciml/breast-cancer-wisconsin-data%5D).

Let's download the data and save it in the local folder with the name data.csv and take a look at it.

data = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/breast-cancer-wisconsin/wdbc.data', header=None)

# specify columns extracted from wbdc.names
data.columns = [
    "id",
    "diagnosis",
    "radius_mean",
    "texture_mean",
    "perimeter_mean",
    "area_mean",
    "smoothness_mean",
    "compactness_mean",
    "concavity_mean",
    "concave points_mean",
    "symmetry_mean",
    "fractal_dimension_mean",
    "radius_se",
    "texture_se",
    "perimeter_se",
    "area_se",
    "smoothness_se",
    "compactness_se",
    "concavity_se",
    "concave points_se",
    "symmetry_se",
    "fractal_dimension_se",
    "radius_worst",
    "texture_worst",
    "perimeter_worst",
    "area_worst",
    "smoothness_worst",
    "compactness_worst",
    "concavity_worst",
    "concave points_worst",
    "symmetry_worst",
    "fractal_dimension_worst",
]

# save the data
data.to_csv("data.csv", sep=",", index=False)

# print the shape of the data file
print(data.shape)

# show the top few rows
display(data.head())

# describe the data object
display(data.describe())

# we will also summarize the categorical field diganosis
display(data.diagnosis.value_counts())

Key observations:
Data has 569 observations and 32 columns.
First field is 'id'.
Second field, 'diagnosis', is an indicator of the actual diagnosis ('M' = Malignant; 'B' = Benign).
There are 30 other numeric features available for prediction.

Create Features and Labels
Split the data into 80% training, 10% validation and 10% testing.

rand_split = np.random.rand(len(data))
train_list = rand_split < 0.8
val_list = (rand_split >= 0.8) & (rand_split < 0.9)
test_list = rand_split >= 0.9

data_train = data[train_list]
data_val = data[val_list]
data_test = data[test_list]

train_y = ((data_train.iloc[:, 1] == "M") + 0).to_numpy()
train_X = data_train.iloc[:, 2:].to_numpy()

val_y = ((data_val.iloc[:, 1] == "M") + 0).to_numpy()
val_X = data_val.iloc[:, 2:].to_numpy()

test_y = ((data_test.iloc[:, 1] == "M") + 0).to_numpy()
test_X = data_test.iloc[:, 2:].to_numpy();

#Use random forest to test the example
from sklearn import ensemble
from sklearn.metrics import roc_auc_score
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score
from sklearn.model_selection import cross_val_score

clf = ensemble.RandomForestClassifier()
accuracy = cross_val_score(clf, train_X, train_y, cv=5, scoring='accuracy') 
print("accuracy:")
print(accuracy)
precision = cross_val_score(clf, train_X, train_y, cv=5, scoring='precision_macro')
print("precision:")
print(precision)
recall = cross_val_score(clf, train_X, train_y, cv=5, scoring='recall_macro')
print("recall:")
print(recall)
f_measure = cross_val_score(clf, train_X, train_y, cv=5, scoring='f1_macro')
print("f_measure:")
print(f_measure)
auc = cross_val_score(clf, train_X, train_y, cv=5, scoring='roc_auc')
print("auc:")
print(auc)
